Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m

```diff
@@ -284,14 +264,17 @@ - (void)toxAV:(OCTToxAV *)toxAV callStateChanged:(OCTToxAVCallState)state friend
     OCTCall *call = [self callFromFriend:friendNumber];
 
     if ((state & OCTToxAVCallStateError) || (state & OCTToxAVCallStateFinished)) {
-        [self logCallAndStopTimer:call type:OCTMessageCallTypeEnd];
-        [self.audioEngine stopAudioFlow:nil];
+        [self.calls updateCall:call updateBlock:^(OCTCall *callToUpdate) {
+            [self logCallAndStopTimer:callToUpdate type:OCTMessageCallTypeEnd];
+        }];
```

Comment:
Why you call this method inside update block? Call isn't actually updated there.